pkg/client/attach.go

```diff
@@ -0,0 +1,257 @@
+package client
+
+import (
+	"context"
+	"io"
+	"net"
+
+	"github.com/containers/conmon-rs/internal/proto"
+	"github.com/containers/podman/v3/libpod/define"
+	"github.com/containers/podman/v3/pkg/kubeutils"
+	"github.com/containers/podman/v3/utils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	AttachPipeStdin  = 1
+	AttachPipeStdout = 2
+	AttachPipeStderr = 3
+)
+
+type TerminalSize struct {
+	Width, Height uint16
+}
+
+type AttachStreams struct {
+	Stdin                                   io.Reader
+	Stdout, Stderr                          io.WriteCloser
+	AttachStdin, AttachStdout, AttachStderr bool
```

Comment:
Could we make those implicit by checking `Stdin/Stdout/Stderr == nil` so we can remove them?

Reply:
How about adding a new structure for them which basically wraps the interface internally?

Reply:
what did you have in mind?

Reply:
> what did you have in mind?

I'll follow-up on that. Let's discuss it there.